posts/2021-07-08-logging-format-21008-beta.adoc

```diff
@@ -0,0 +1,240 @@
+
+Applications are expected to provide health check procedures by implementing the HealthCheck interface with the @startup, `@Liveness`, or `@readiness` annotations. These are used by Open Liberty to verify the Startup, Liveness, or Readiness of the application, respectively. Add the logic of your health check in the `call()` method, and return the `HealthCheckResponse` object, by using the simple `up()`/`down()` methods from the API:
+
+[source, java]
+----
+**Startup Check**
+@Startup
+@ApplicationScoped
+public class StartupMemUsageCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       if (getMemUsage() < 0.95) {
+           return HealthCheckResponse.up("memUsage");
+       }
+       return HealthCheckResponse.down("memUsage");
+       ...
+     }
+}
+----
+
+To view the status of each health check, accessthe `http://<hostname>:<port>/health/started` endpoint.
+
+
+To enable the new beta features in your app, add them to your `server.xml`:
+
+[source, xml]
+----
+<feature>mpHealth-3.1</feature>
+----
+
+To find out more viist:
+
+* link:https://github.com/eclipse/microprofile-health/releases/tag/3.1-RC1[MicroProfile Health Check 3.1 Release Page]
+* link:https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/release_notes.asciidoc[Release Notes]
+
+[#tbasic]
+=== TBASIC Logging Format Beta
+
+A new logging format, `TBASIC`, is added to match the Websphere Application Server basic logging format. This new format option can be used for `consoleFormat`, `messageFormat`, or `traceFormat`. `TBASIC` format matches the existing `trace BASIC` format (so now either `TBASIC` or `BASIC` can be used to refer to this format for trace). Using this new format option will allow users that utilize log parsers that work with the Websphere Application Server basic format to use the same parsers with Open Liberty logs.
+
+The new options can be used in the bootstrap.properties file:
+
+[source, java]
+----
+com.ibm.ws.logging.message.format=tbasic
+com.ibm.ws.logging.console.format=tbasic
+com.ibm.ws.logging.trace.format=tbasic
+----
+
+You can also change the format by editing the server.env and adding the following lines:
+
+[source, java]
+----
+WLP_LOGGING_MESSAGE_FORMAT=TBASIC
+WLP_LOGGING_CONSOLE_FORMAT=TBASIC
+----
+
+
+TBASIC Logs Example:
+
+[source, java]
+----
+[24/03/21 15:04:10:331 EDT] 00000001 FrameworkMana A   CWWKE0001I: The server defaultServer has been launched.
+[24/03/21 15:04:11:338 EDT] 00000001 FrameworkMana I   CWWKE0002I: The kernel started after 1.177 seconds
+[24/03/21 15:04:11:465 EDT] 0000003e FeatureManage I   CWWKF0007I: Feature update started.
+[24/03/21 15:04:11:635 EDT] 00000033 DropinMonitor A   CWWKZ0058I: Monitoring dropins for applications.
+----
+
+
+[#staleConnectionID]
+=== Customizing Stale Connection Identification
```

Comment:
This section needs to be a bit clearer about what the "new" function is. `identifyException` is already a GA config element - https://openliberty.io/docs/21.0.0.6/reference/config/identifyException.html

Reply:
Seems like Nathan will be out all week this week.

Reply:
we should try to follow up with someone else then